presto-orc/src/main/java/com/facebook/presto/orc/writer/TimestampColumnWriter.java

```diff
@@ -131,10 +131,22 @@ public void writeBlock(Block block)
                 // zero, because many systems, like Presto, only record millisecond or microsecond precision
                 // timestamps. To optimize storage, if the value has more than two trailing zeros, the trailing
                 // decimal zero digits are removed, and the last three bits are used to record how many zeros
-                // were removed (minus one).
+                // were removed (minus one):
+                // +-------------+---------------+
+                // | Last 3 Bits | # Trailing 0s |
+                // +-------------+---------------+
+                // |    0b000    |       0       |
+                // |    0b001    |       2       |
+                // |    0b010    |       3       |
+                // |    0b011    |       4       |
+                // |    0b100    |       5       |
+                // |    0b101    |       6       |
+                // |    0b110    |       7       |
+                // |    0b111    |       8       |
+                // +-------------+---------------+
```

Comment:
What about 
```
                //   # Trailing 0s   Last 3 Bits   Example nanos       Example encoding
                //         0            0b000        123456789     (123456789 << 3) | 0b000
                //         1            0b000        123456780     (123456780 << 3) | 0b000
                //         2            0b001        123456700      (2345678 << 3) | 0b001
                //         3            0b010      
                //         4            0b011      
                //         5            0b100      
                //         6            0b101      
                //         7            0b110      
                //         8            0b111      
                //
                // In Presto, we only have millisecond precision. Therefore, we always use the encoding for 6 trailing zeros (except when input is zero).
                // For simplicity, we don't dynamically use 6, 7, 8 depending on the circumstance.

Reply:
@haozhun is there a typo in the third example encoding?  Should it be `(1234567 << 3) | 0b001`?